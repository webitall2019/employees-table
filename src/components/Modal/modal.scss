
.modal{
    display: none;
    width: 100%;
    height: 100%;
    position: absolute;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: rgba(60, 71, 71, 0.507);
    &.visible{
        display: block;
    }
    &__btn{
        &--close{
            position: absolute;
            right: 20%;
            top: 20%;
            width: 50px;
            height: 50px;
            border-radius: 50%;
            background-color: lightcoral;
            span{
                display: block;
                width: 30px;
                height: 2px;
                background-color: #000;
                margin: 0 auto;
                transform: rotate(45deg);
                
                &:first-child{
                    position: relative;
                    top: 2px;
                    transform: rotate(-45deg);
                }
            }
            &:hover{
                transition: ease .3s;
                transform: rotateZ(90deg);
               
            }
        }
    }
}

