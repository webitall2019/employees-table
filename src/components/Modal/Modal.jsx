import React from "react";
import "./modal.scss";

const Modal = ({ children, modalHandler }) => (
  <div className="modal" role="showModal" data-testid="modalTestId">
    <button onClick={modalHandler} className="modal__btn--close">
      <span></span>
      <span></span>
    </button>
    {children}
  </div>
);

export default Modal;
