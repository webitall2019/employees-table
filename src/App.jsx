import React, { useState } from "react";
import "./app.scss";
import BarChart from "./components/BarChart/BarChart";
import PieChart from "./components/PieChart/PieChart";
import Modal from "./components/Modal/Modal";
import Table from "./components/Table/Table";
import NewEmployee from "./components/NewEmployee/NewEmployee";

const App = () => {
  const [isOpen, setIsOpen] = useState(false);
  const modalHandler = () => setIsOpen(prevState => !prevState);
  
  return (
    <div className="app">
      <div className="container">
        <button onClick={modalHandler} className="app__add-user">
          Add new Employee
        </button>
        <Table />
        <div className="charts">
          <PieChart />
          <BarChart />
        </div>
        {isOpen && (
          <Modal modalHandler={modalHandler}>
            <NewEmployee />
          </Modal>
        )}
      </div>
    </div>
  );
};

export default App;

