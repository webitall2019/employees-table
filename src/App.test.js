import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import App from "./App";
import store from "./store/store";
import Modal from "./components/Modal/Modal";
Modal;

describe("<App/> component", () => {
  it("should match snapshot WHEN rendered", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <App>
          <Modal />
        </App>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
