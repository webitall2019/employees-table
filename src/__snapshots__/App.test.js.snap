// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> component should match snapshot WHEN rendered 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="container"
    >
      <button
        class="app__add-user"
      >
        Add new Employee
      </button>
      <div
        class="table"
      >
        <div
          class="table__header"
        >
          <div
            class="row"
            data-testid="titleRender"
          />
        </div>
        <div
          class="table__body"
        />
      </div>
      <div
        class="charts"
      >
        <div
          class="pie-chart chart"
          data-testid="pieChart"
        />
        <div
          class="bar-chart chart"
          data-testid="barChart"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
